frontend/src/pages/adminRoles.js

```diff
@@ -1,64 +1,201 @@
 import React from 'react'
-import { Button, Table } from 'reactstrap'
+import {
+  Button,
+  Table,
+  Card,
+  CardBody,
+  Modal,
+  ModalBody,
+  ModalHeader,
+  ModalFooter,
+  FormGroup,
+  Input,
+  Label
+} from 'reactstrap'
 import ChangeRole from '../components/changeRole'
-import ChangeYear from '../components/changeYear'
 import Head from '../components/head'
 import Nav from '../components/nav'
-
-const SAMPLE_MEMBER_ID = 1234
+import { getAllUsers, updateUserRole, updateServerUserRole } from '../utils/api'
 
 class AdminRoles extends React.Component {
   constructor(props) {
     super(props)
     this.state = {
-      isEditing: false
+      isEditing: false,
+      newRole: -1,
+      selectedUser: -1,
+      users: [],
+      adminPassword: '',
+      showPasswordModal: false,
+      error: ''
     }
   }
 
+  componentDidMount = () => {
+    this.getUsers()
+  }
+
+  getUsers = () => {
+    getAllUsers().then(resp => {
+      this.setState({ users: resp.result })
+    })
+  }
+
+  handleRoleChange = (e, idx) => {
+    e.persist() // not sure why this is needed?
+    this.setState({ newRole: e.target.value, selectedUser: idx })
+  }
+
+  handleRoleSubmit = userEmail => {
+    if (this.state.adminPassword === '') {
+      this.setState({ showPasswordModal: true })
+    }
+  }
+
+  handleChange = event => {
+    const value = event.target.value
+    const name = event.target.name
+    this.setState({ [name]: value })
+  }
+
+  handlePasswordComplete = () => {
+    const { users, selectedUser, newRole, adminPassword } = this.state
+
+    updateServerUserRole(users[selectedUser].email, newRole, adminPassword).then(resp => {
+      if (resp.status != 400) {
```

Comment:
to clarify, the PI server doesn't return a success value so using the status instead